pkg/monitoring/producer.go

```diff
@@ -0,0 +1,65 @@
+package monitoring
```

Comment:
This and many files in this implementation are currently or are close to being Solana agnostic.

As this monitoring service will be useful across networks and across protocols (contracts) I'd suggest as a first step we try to decouple the two best we can ATM.

- `pkg/monitoring` - generic monitoring code
- `pkg/monitoring/solana/...` - solana specific implementations and details

Then we iterate on the monitoring interfaces couple of times until we get a fit, and we can raise the generic part to the Relay SDK level, which all relays use to implement and bootstrap their monitoring serivce.